Dashboard

rick-astral-cat opened the pull request:
This commit add necessary information and retrieve data from database to dashboard.
This show:
- Summary boxes with statistics
- Last session chart
- Main navigation bars

rick-astral-cat committed:
[ADD] ApexCharts dependency and configuration

rick-astral-cat committed:
[ADD] Blade service provider for custom directives

Running in dev mode don't forget run: php artisan view:clear
And restart your IDE to update blade directives

rick-astral-cat committed:
[ADD] Session service and dashboard controller

rick-astral-cat committed:
[ADD] Last session Apexchart on dashboard with dark mode support

rick-astral-cat committed:
[ADD] Summary boxes on dashboard

rick-astral-cat committed:
[IMP] Style for header name

rick-astral-cat committed:
[IMP] Light mode styles for dashboard

rick-astral-cat committed:
[ADD] Navigation menus for main views

rick-astral-cat committed:
[ADD] Boards service class

rick-astral-cat committed:
[ADD] Profiles service class

rick-astral-cat committed:
[ADD] Sessions count method

rick-astral-cat committed:
[IMP] Show box summaries data from database

crazybolillo commented:
https://stackoverflow.com/a/59991727
https://github.com/BookStackApp/BookStack/blob/development/app/Entities/Models/Book.php

crazybolillo commented on app/Http/Controllers/Dashboard.php:
```diff
@@ -0,0 +1,51 @@
+        $session = $this->sessionService->lastSession();
+
+        //Extract data for X and Y Axis
+        if (! empty($session)) {
```
You need to handle the case when sessions are empty. This is why the lints are failing. If session is empty,  xAxis and yAxis won't be defined.

crazybolillo commented on app/Http/Controllers/Dashboard.php:
```diff
@@ -0,0 +1,51 @@
+        $this->profileService = new ProfileService();
+    }
+
+    //Method to show dashboard and send necessary information
```
```suggestion
    /**
    * Method to show dashboard and send necessary information
    */
```
Single line comments like that are generally not used for functions, they also don't show up on IDEs when you search the docs for a method.

crazybolillo commented on app/Models/Session.php:
```diff
@@ -8,4 +8,10 @@
 {
     use HasFactory;
+
+    //Relation to measurements for session
```
Same here, try using `/** */`

crazybolillo reviewed (dismissed):
Looks good, you just need to follow the linter suggestions, which mostly consists of adding return types. Once you do that all errors should go on their own.

crazybolillo committed:
Fix lints, correct code style

This commit fixes the linter errors generated by the previous commit. It
also makes some changes to code style.

* Set default values for xAxis, yAxis
* Use multiline comments for function docstrings
* Add return types to functions where required by phpstan
